src/Model/GroundWaterEnergy/gwe-est.f90

```diff
@@ -262,37 +266,75 @@ subroutine est_fc_dcy(this, nodes, cold, cnew, nja, matrix_sln, &
       vcell = this%dis%area(n) * (this%dis%top(n) - this%dis%bot(n))
       swtpdt = this%fmi%gwfsat(n)
       !
-      ! -- add decay rate terms to accumulators
-      idiag = this%dis%con%ia(n)
-      if (this%idcy == 1) then
+      ! -- add zero-order decay rate terms to accumulators
+      if (this%idcy == 2 .and. (this%idcysrc == 1 .or. &
```

Comment:
One thing you might consider is using an enumeration instead of integers for this.  I think we have all been transitioning that way, and it would make this statement much easier to digest.

```
if (this%idcy == DECAY_ZERO .and. (this%idcysrc == DECAY_AQUEOUS_ONLY .or. this%idcysrc == DECAY_AQUEOUS_SOLD))
```

Reply:
Suggested change implemented with b7a2330